src/commands/scaling/deleteScaleRule/DeleteScaleRuleStep.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { nonNullProp } from "@microsoft/vscode-azext-utils";
+import { ext } from "../../../extensionVariables";
+import { RevisionsItemModel } from "../../../tree/revisionManagement/RevisionItem";
+import { localize } from "../../../utils/localize";
+import { getParentResource } from "../../../utils/revisionDraftUtils";
+import { RevisionDraftUpdateBaseStep } from "../../revisionDraft/RevisionDraftUpdateBaseStep";
+import { IAddScaleRuleContext } from "../addScaleRule/IAddScaleRuleContext";
+import { IDeleteScaleRuleContext } from "./IDeleteScaleRuleContext";
+
+export class DeleteScaleRuleStep<T extends IAddScaleRuleContext> extends RevisionDraftUpdateBaseStep<T> {
+    public priority: number = 100;
+
+    constructor(baseItem: RevisionsItemModel) {
+        super(baseItem);
+    }
+
+    public async execute(context: IDeleteScaleRuleContext): Promise<void> {
+        this.revisionDraftTemplate.scale ||= {};
+        this.revisionDraftTemplate.scale.rules ||= [];
+
+        const index = this.revisionDraftTemplate.scale.rules.indexOf(nonNullProp(context, 'scaleRule'));
```

Comment:
The idea here is spot on, but due to how objects are compared with Node (by reference/address), this will probably always return `-1`.  

To circumvent this, a likely better way to match would be to compare one of their properties instead.  Something like this could work:
`const index = this.revisionDraftTemplate.scale.rules.findIndex(r => r.name === nonNullValueAndProp(context.scaleRule, 'name'));`

I would also maybe add an if statement guard clause immediately after for the case when index is `-1` (no match) and handle it by throwing an error or something 🤔